src/main/kotlin/com/autonomousapps/tasks/classAnalysisTasks.kt

```diff
@@ -119,31 +124,33 @@ abstract class JarAnalysisTask @Inject constructor(
 
 interface JarAnalysisParameters : WorkParameters {
   val variantFiles: SetProperty<VariantFile>
-  var jar: File
-  var kaptJavaSource: Set<File>
-  var layouts: Set<File>
-  var testFiles: Set<File>
-  var report: File
-  var reportPretty: File
+  var jar: RegularFileProperty
+  var kaptJavaSource: FileCollection
+  var layouts: FileCollection
+  var testJavaClassesDir: DirectoryProperty
+  var testKotlinClassesDir: DirectoryProperty
+  var report: RegularFileProperty
+  var reportPretty: RegularFileProperty
 }
 
 abstract class JarAnalysisWorkAction : WorkAction<JarAnalysisParameters> {
 
   private val logger = getLogger<JarAnalysisTask>()
 
   override fun execute() {
+    val testFiles = getTestFiles(parameters.testJavaClassesDir, parameters.testKotlinClassesDir)
```

Comment:
why is it important to wait until we're in the work action to do this? And if it _is_ more correct to do it here, probably make the function available somewhere else (besides the task class)?